ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound.hs

```diff
@@ -321,10 +319,16 @@ txSubmissionInbound _tracer maxUnacked mpReader mpWriter _version =
             bufferedTxs'' = foldl' (flip Map.delete)
                                    bufferedTxs' acknowledgedTxIds
 
-        _ <- mempoolAddTxs txsReady
+        _writtenTxids <- mempoolAddTxs txsReady
+
+        -- If we are acknowleding transactions that are still in unacknowledgedTxIds'
+        -- we need to re-add them so that we also can acknowledge them again later.
+        -- This will happen incase of duplicate txids within the same window.
+        let live =  filter (`elem` unacknowledgedTxIds') $ toList acknowledgedTxIds
+            !bufferedTxs''' = bufferedTxs'' <> (Map.fromList (zip  live (repeat Nothing)))
```

Comment:
```suggestion
            !bufferedTxs''' = bufferedTxs'' <> Map.fromList (zip live (repeat Nothing))
```
Also, the moment we need three primes (`'''`) might be a sign that we should switch to a different naming scheme. Either by numbering or by using different, more descriptive names.

Reply:
I am more and more leaning toward using unicode `bufferedTxs₃` (this makes `agda` code easier to read - though harder to type).

Reply:
I'll figure out a better naming scheme that doesn't require unicode.

Reply:
The revolution is still to happen :D ! Unicode has its WATs: how's that possible that there's not supscript `b,c,d`.